bokehjs/src/coffee/document.coffee

```diff
@@ -650,8 +651,13 @@ export class Document
             throw new Error("Cannot apply patch to #{patched_id} which is not in the document")
           patched_obj = @_all_models[patched_id]
           attr = event_json['attr']
-          value = Document._resolve_refs(event_json['new'], old_references, new_references)
-          patched_obj.setv({ "#{attr}" : value })
+          model_type = event_json['model']['type']
+          if attr == 'data' and model_type == 'ColumnDataSource'
+            [data, shapes] = decode_column_data(event_json['new'])
+            patched_obj.setv({ "_shapes": shapes, "data" : data}, {notify: false})
```

Comment:
Without this change it would have to re-encode the typed arrays so just so that ``ClientSession._should_suppress_on_change`` could compare and suppress them correctly.